Persist OO Style choice (& Change OO Style Architecture)

subhramit opened the pull request:
<!-- 
Describe the changes you have made here: what, why, ... 
Link the issue that will be closed, e.g., "Closes #333".
If your PR closes a koppor issue, link it using its URL, e.g., "Closes https://github.com/koppor/jabref/issues/47".
"Closes" is a keyword GitHub uses to link PRs with issues; do not change it.
Don't reference an issue in the PR title because GitHub does not support auto-linking there.
-->
Grounds for upcoming preferences PR (https://github.com/subhramit/jabref/pull/6)
For the final PR, visit https://github.com/JabRef/jabref/pull/11521
### Mandatory checks

<!-- 
- Go through the list below. Please don't remove any items.
- [x] done; [ ] not done / not applicable
-->

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

subhramit committed:
Rename OOBibStyle to JStyle

subhramit committed:
Rename OOBibStyle to JStyle (Other classes)

subhramit committed:
Separation of Style Type classes

subhramit committed:
Rename OO Style file in preferences (to be replaced later)

Siedlerchr commented on src/main/java/org/jabref/preferences/JabRefPreferences.java:
```diff
@@ -354,7 +354,7 @@ public class JabRefPreferences implements PreferencesService {
     public static final String OO_SYNC_WHEN_CITING = "syncOOWhenCiting";
     public static final String OO_USE_ALL_OPEN_BASES = "useAllOpenBases";
-    public static final String OO_BIBLIOGRAPHY_STYLE_FILE = "ooBibliographyStyleFile";
+    public static final String JSTYLE_FILE = "jStyleFile";
```
Please do not rename the preferences, as otherwise existing users will have problems. You would need to provide a migration otherwise

subhramit replied:
Reverted

subhramit committed:
Undo rename preferences

subhramit committed:
Extend OO Preferences to include current style

subhramit committed:
Add getPath to interface methods

subhramit committed:
Use concrete object in StyleSelectDialogViewModel

subhramit committed:
Separate unified and csl style seelction

subhramit committed:
Attempt - Unified style

koppor committed:
Fix Chocolate.bib (#11491)

gradle-update-robot committed:
Update Gradle Wrapper from 8.8 to 8.9.

Signed-off-by: gradle-update-robot <gradle-update-robot@regolo.cc>

Siedlerchr committed:
Merge pull request #11494 from JabRef/gradlew-update-8.9

Update Gradle Wrapper from 8.8 to 8.9

subhramit committed:
FAQ updates (#11486)

* Update FAQs

* Remove trailing space

* Apply review changes

* Update docs/code-howtos/faq.md

---------

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

dependabot[bot] committed:
Bump org.openrewrite.recipe:rewrite-recipe-bom from 2.14.0 to 2.15.0 (#11495)

Bumps [org.openrewrite.recipe:rewrite-recipe-bom](https://github.com/openrewrite/rewrite-recipe-bom) from 2.14.0 to 2.15.0.
- [Release notes](https://github.com/openrewrite/rewrite-recipe-bom/releases)
- [Commits](https://github.com/openrewrite/rewrite-recipe-bom/compare/v2.14.0...v2.15.0)

---
updated-dependencies:
- dependency-name: org.openrewrite.recipe:rewrite-recipe-bom
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump com.kohlschutter.junixsocket:junixsocket-mysql from 2.9.1 to 2.10.0 (#11496)

Bumps [com.kohlschutter.junixsocket:junixsocket-mysql](https://github.com/kohlschutter/junixsocket) from 2.9.1 to 2.10.0.
- [Release notes](https://github.com/kohlschutter/junixsocket/releases)
- [Commits](https://github.com/kohlschutter/junixsocket/compare/junixsocket-2.9.1...junixsocket-2.10.0)

---
updated-dependencies:
- dependency-name: com.kohlschutter.junixsocket:junixsocket-mysql
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jsoup:jsoup from 1.17.2 to 1.18.1 (#11497)

Bumps [org.jsoup:jsoup](https://github.com/jhy/jsoup) from 1.17.2 to 1.18.1.
- [Release notes](https://github.com/jhy/jsoup/releases)
- [Changelog](https://github.com/jhy/jsoup/blob/master/CHANGES.md)
- [Commits](https://github.com/jhy/jsoup/compare/jsoup-1.17.2...jsoup-1.18.1)

---
updated-dependencies:
- dependency-name: org.jsoup:jsoup
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump commons-logging:commons-logging from 1.3.2 to 1.3.3 (#11499)

Bumps commons-logging:commons-logging from 1.3.2 to 1.3.3.

---
updated-dependencies:
- dependency-name: commons-logging:commons-logging
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump com.kohlschutter.junixsocket:junixsocket-core from 2.9.1 to 2.10.0 (#11498)

Bumps [com.kohlschutter.junixsocket:junixsocket-core](https://github.com/kohlschutter/junixsocket) from 2.9.1 to 2.10.0.
- [Release notes](https://github.com/kohlschutter/junixsocket/releases)
- [Commits](https://github.com/kohlschutter/junixsocket/compare/junixsocket-2.9.1...junixsocket-2.10.0)

---
updated-dependencies:
- dependency-name: com.kohlschutter.junixsocket:junixsocket-core
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump gittools/actions from 1.1.1 to 1.2.0 (#11500)

Bumps [gittools/actions](https://github.com/gittools/actions) from 1.1.1 to 1.2.0.
- [Release notes](https://github.com/gittools/actions/releases)
- [Commits](https://github.com/gittools/actions/compare/v1.1.1...v1.2.0)

---
updated-dependencies:
- dependency-name: gittools/actions
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump src/main/resources/csl-styles from `b2be5ae` to `fd6cb3e` (#11501)

Bumps [src/main/resources/csl-styles](https://github.com/citation-style-language/styles) from `b2be5ae` to `fd6cb3e`.
- [Release notes](https://github.com/citation-style-language/styles/releases)
- [Commits](https://github.com/citation-style-language/styles/compare/b2be5aeeee7f00fd2032ac1daad995bbe95398cf...fd6cb3e81762055d107839c3c288c359985f81c8)

---
updated-dependencies:
- dependency-name: src/main/resources/csl-styles
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Siedlerchr commented on src/main/java/org/jabref/gui/openoffice/StyleSelectItemViewModel.java:
```diff
@@ -44,15 +48,24 @@ public StringProperty fileProperty() {
+    }
+
+    public OOStyle getStyleUnified() {
+        if (ooStyle instanceof JStyle jStyle) {
```
this casting is not needed right? You always get a OOStyle back

Siedlerchr commented:
codewise goes already in the right direction

Siedlerchr commented on src/main/java/org/jabref/preferences/JabRefPreferences.java:
```diff
@@ -1392,6 +1399,33 @@ public GroupsPreferences getGroupsPreferences() {
+            return style;
+        }
+
+        String currentStyle = get(OO_CURRENT_STYLE);
```
Here, you now need to do a check based on the value in the preferences:
If it ends with .csl  then you have a CSL style 
  if (CitationStyle.isCitationStyleFile(currentStyle)) {   
CitationStyle.createCitationStyleFromFile(currentStyle)

if( isJStyle(currentStyle) )
ooBibStyleFromjstyle.

subhramit replied:
Okay, makes sense now. I tried to store the styletype object property, whereas I had to use a string and deduce the type from that.

subhramit committed:
CSL4LibreOffice - A [GSoC '24] (#11477)

* Finish CSL Style selection Tab

* Preview Section (To Diagnose)

* Rename preview boxes

* Fixed CSL Style Preview as per suggestions

* Resize CSL Prveiw box height, Comply with Checkstyle

* IntelliJ refactoring & Add back filter functionality

* Draft Commit: Establish Basic Framework

* Draft Commit: Establish Basic Framework

* Connect UI with CSL Logic (Phase 1), Improve HTML Formatter (Phase 2), Cleanup

* Add logging of Cited Style Name for Testing

* Add preliminary test support for in-text citations

* Refine Implementation: Refactoring [1]

* Improve Logging

* Major CSL Formatting improvements

* Fix test, OpenRewrite

* Remove in-text citation test

* Fix architecture test: Move business logic from model to logic directory

* Fix order of imports

* Extend CSL Citation feature to manually selected library entries

* Don't assume Databasecontext

* Major bug fix: Using Filter Seelcted entries from start (Use name instead of index, basis for preferences)

* Implement extensible selection of Style categories, Reconnect JStyle pipeline

* Proper notification of selected style

* Separation of bibliographic and in-text citation handling

* Log instead of throwing exceptions while citing

* Handle the case of margins (gap between citation number and text)

* Refactoring, Localization

* Changelog

* RewriteRun

* Add ADR for Style Type Selection Problem

* Remove unused localization

* Fix markdown error

* Remove comment

* remove csl submodules

* Resolve submodules

* gitmodules

* Recover gitmodules

* recover gitmodules

* Recovery attempt

* fix submodules

* Add double click to select CSL style functionality

* Add double-click to select CSL style feature

* Adjust changelog

* Optimize transformHtml

* Add parameterized logging

* Remove thrown exception, replace by logger

* Add a return statement (enforced)

* Remove unused fields

* openRewrite

* Remove boilerplate comment

---------

Co-authored-by: Siedlerchr <siedlerkiller@gmail.com>

koppor committed:
Add missing issue numbers

Siedlerchr committed:
Merge pull request #11503 from JabRef/koppor-patch-1

Add missing issue numbers

Siedlerchr committed:
New Crowdin updates (#11504)

* New translations jabref_en.properties (French)

* New translations jabref_en.properties (Spanish)

* New translations jabref_en.properties (Arabic)

* New translations jabref_en.properties (Danish)

* New translations jabref_en.properties (German)

* New translations jabref_en.properties (Greek)

* New translations jabref_en.properties (Finnish)

* New translations jabref_en.properties (Italian)

* New translations jabref_en.properties (Japanese)

* New translations jabref_en.properties (Korean)

* New translations jabref_en.properties (Dutch)

* New translations jabref_en.properties (Norwegian)

* New translations jabref_en.properties (Polish)

* New translations jabref_en.properties (Portuguese)

* New translations jabref_en.properties (Russian)

* New translations jabref_en.properties (Swedish)

* New translations jabref_en.properties (Turkish)

* New translations jabref_en.properties (Ukrainian)

* New translations jabref_en.properties (Chinese Simplified)

* New translations jabref_en.properties (Chinese Traditional)

* New translations jabref_en.properties (Vietnamese)

* New translations jabref_en.properties (Portuguese, Brazilian)

* New translations jabref_en.properties (Indonesian)

* New translations jabref_en.properties (Persian)

* New translations jabref_en.properties (Tagalog)

koppor committed:
Fix handling of relative-file storage and auto linking (#11492)

* Fix handling of relative-file storage and auto linking

* Update CHANGELOG.md

* Adapt tests to new behavior

* Refine code

* Use LinkedHashSet instead of List for building the resulting list

odneill committed:
Add minimal support for biblatex data annotations (#11506)

* Add minimal support for biblatex data annotations

Fixes #11505

* update changelog

kataramarina committed:
Save unlinked local files dialog prefs (#11493)

* Added saving of selected options for unlinked local files dialog (#11493)

* Update CHANGELOG.md

* Removed not needed checkbox from dialog, resolved comments

---------

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

koppor committed:
Switch to stream-based loading (#11479)

* Switch to stream-based loading

* Fix typos

* Add architecture test

* Fix ProtectedTermsLoaderTest and minor refactor

* Remove "static"

* Introduce AllowedToUseClassGetResource - and do some rewrite

* Streamline FileUtil.copy

* Rewrite CitationStyle.java more to streams

* More change to asStream

* Convert to getResourceAsStream

* Remoe unused imports

* Fix MathMLParser

* Remove imports

* Fix test

---------

Co-authored-by: Carl Christian Snethlage <50491877+calixtus@users.noreply.github.com>

LoayGhreeb committed:
Fix NPE when saving preferences (#11509)

* Fix NPE when saving preferences

* Update CHANGELOG.md

LoayGhreeb committed:
Always enable 'Check for Updates' menu button (#11485)

* Store version check setting in internal preferences

* Update version check text

* Always enable 'Check for Updates' menu button

* CHANGELOG

* Update src/main/resources/l10n/JabRef_en.properties

Co-authored-by: Carl Christian Snethlage <50491877+calixtus@users.noreply.github.com>

* Update NetworkTab.fxml

---------

Co-authored-by: Carl Christian Snethlage <50491877+calixtus@users.noreply.github.com>
Co-authored-by: Christoph <siedlerkiller@gmail.com>

Siedlerchr committed:
New Crowdin updates (#11514)

* New translations jabref_en.properties (French)

* New translations jabref_en.properties (Spanish)

* New translations jabref_en.properties (German)

* New translations jabref_en.properties (Italian)

* New translations jabref_en.properties (Dutch)

* New translations jabref_en.properties (Polish)

* New translations jabref_en.properties (Turkish)

* New translations jabref_en.properties (Chinese Simplified)

* New translations jabref_en.properties (Portuguese, Brazilian)

Siedlerchr committed:
import PMID field in Pubmed (#11513)

* import PMID field in Pubmed

Fixes https://github.com/JabRef/jabref/issues/11488

* fix tests

---------

Co-authored-by: Carl Christian Snethlage <50491877+calixtus@users.noreply.github.com>

InAnYan committed:
Add OpenAI privacy policy (#11511)

* Add OpenAI privacy policy

* Fix indentation

* Fix OpenAI privacy link

---------

Co-authored-by: Carl Christian Snethlage <50491877+calixtus@users.noreply.github.com>

Siedlerchr committed:
New translations jabref_en.properties (Portuguese, Brazilian) (#11516)

LoayGhreeb committed:
Fix NPE when saving preferences (#11515)

* Revert "Fix NPE when saving preferences (#11509)"

This reverts commit ea00fe62849c1a69b1b86a54ac55fae092520bbf.

* Fix NPE when saving preferences

* Update CHANGELOG.md

koppor committed:
Try to use "release" (made in fork) (#11517)

subhramit committed:
Review changes - I

subhramit committed:
Clean unused imports

subhramit committed:
Adapt StyleSelectDialogViewModel

Siedlerchr committed:
Merge remote-tracking branch 'upstream/main' into oo-architecture

* upstream/main:
  Fix NPE when saving preferences (#11509)
  Switch to stream-based loading (#11479)
  Save unlinked local files dialog prefs (#11493)
  Add minimal support for biblatex data annotations (#11506)
  Fix handling of relative-file storage and auto linking (#11492)
  New Crowdin updates (#11504)
  Add missing issue numbers
  CSL4LibreOffice - A [GSoC '24] (#11477)
  Bump src/main/resources/csl-styles from `b2be5ae` to `fd6cb3e` (#11501)
  Bump gittools/actions from 1.1.1 to 1.2.0 (#11500)
  Bump com.kohlschutter.junixsocket:junixsocket-core from 2.9.1 to 2.10.0 (#11498)
  Bump commons-logging:commons-logging from 1.3.2 to 1.3.3 (#11499)
  Bump org.jsoup:jsoup from 1.17.2 to 1.18.1 (#11497)
  Bump com.kohlschutter.junixsocket:junixsocket-mysql from 2.9.1 to 2.10.0 (#11496)
  Bump org.openrewrite.recipe:rewrite-recipe-bom from 2.14.0 to 2.15.0 (#11495)
  FAQ updates (#11486)
  Update Gradle Wrapper from 8.8 to 8.9.
  Fix Chocolate.bib (#11491)

# Conflicts:
#	src/main/java/org/jabref/gui/openoffice/OOBibBase.java
#	src/main/java/org/jabref/gui/openoffice/OpenOfficePanel.java
#	src/main/java/org/jabref/gui/openoffice/StyleSelectDialogView.java
#	src/main/java/org/jabref/gui/openoffice/StyleSelectDialogViewModel.java
#	src/main/java/org/jabref/preferences/JabRefPreferences.java

Siedlerchr committed:
fix prefs

subhramit committed:
Fix style checks

Siedlerchr committed:
fix compile errors

Siedlerchr committed:
store prefs

Siedlerchr committed:
fix selection of style in preferenmes

Siedlerchr committed:
add todo

Siedlerchr committed:
Update to javafx 22.02 (#11518)

https://github.com/openjdk/jfx22u/blob/master/doc-files/release-notes-22.0.2.md

Siedlerchr committed:
Always show star for local unsaved library (#11519)

* Always show star for local unsaved library

* changelog

subhramit committed:
Debug phase 1

subhramit committed:
Bug fixes 2 - defaulting jstyles

subhramit committed:
Clean preference implementation, add comments

subhramit committed:
Bug fix -3, more cleanup

subhramit committed:
Bug fix -4 (Failing viewloader)

subhramit committed:
Refactoring, Documentation

subhramit committed:
Add currently set style label

subhramit committed:
Remove unused setStyleProperty function

subhramit committed:
Separate blocks for JStyle and CSL

subhramit committed:
Merge branch 'main' of https://github.com/JabRef/jabref into oo-architecture
